web/src/components/Header.tsx

```diff
@@ -0,0 +1,66 @@
+import { Button } from "@/components/njwds-extended/Button";
+import { AuthContext } from "@/contexts/authContext";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import analytics from "@/lib/utils/analytics";
+import { templateEval } from "@/lib/utils/helpers";
+import Config from "@businessnjgovnavigator/content/fieldConfig/config.json";
+import { getCurrentDateInNewJersey } from "@businessnjgovnavigator/shared/";
+import { useRouter } from "next/router";
+import { useContext } from "react";
+
+export const Header = () => {
+  const { state } = useContext(AuthContext);
+
+  const { userData } = useUserData();
+  const router = useRouter();
+
+  const editOnClick = () => {
+    analytics.event.roadmap_profile_edit_button.click.go_to_profile_screen();
+    router.push("/profile");
+  };
+
+  const getHeader = (): string => {
+    return userData?.user.name
+      ? templateEval(Config.roadmapDefaults.roadmapTitleTemplateForUserName, {
+          name: userData.user.name,
+        })
+      : Config.roadmapDefaults.roadmapTitleBusinessAndUserMissing;
+  };
+
+  const getButtonText = () => {
+    let buttonText;
+
+    if (state.isAuthenticated === "FALSE" || state.isAuthenticated === "UNKNOWN")
+      buttonText = Config.roadmapDefaults.guestModeToProfileButtonText;
+
+    if (
+      (userData?.profileData.businessName === undefined || userData?.profileData.businessName === "") &&
+      state.isAuthenticated === "TRUE"
+    )
+      buttonText = Config.roadmapDefaults.genericToProfileButtonText;
+
+    if (userData?.profileData.businessName && state.isAuthenticated === "TRUE")
+      buttonText = userData.profileData.businessName;
+
+    return buttonText;
```

Comment:
I think this would be cleaner if we just returned the value in the if-block instead of setting a variable and returning at the end? It would also prevent the possibility of accidentally changing a value if multiple if-statements apply to a situation